Silksong.ModMenu.csproj

```diff
@@ -24,21 +24,46 @@
     <PackageReadmeFile>README.md</PackageReadmeFile>
     <PackageIcon>icon.png</PackageIcon>
     <RepositoryUrl>https://github.com/silksong-modding/Silksong.ModMenu</RepositoryUrl>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <ManagedFolder Condition="$(SilksongFolder) != ''"
+      >$(SilksongFolder)\Hollow Knight Silksong_Data\Managed</ManagedFolder
+    >
+    <!-- Set to true to use local references for development purposes. -->
+    <UseLocalRefs>false</UseLocalRefs>
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="BepInEx.Analyzers" Version="1.0.8" PrivateAssets="all" />
     <PackageReference Include="BepInEx.Core" Version="5.4.21" />
     <PackageReference Include="HarmonyX" Version="2.9.0" />
     <PackageReference Include="Hamunii.BepInEx.AutoPlugin" Version="2.0.1" PrivateAssets="all" />
     <PackageReference Include="Microsoft.Unity.Analyzers" Version="1.25.0" PrivateAssets="all" />
+    <PackageReference Include="MonoDetour" Version="0.7.5" />
+    <PackageReference Include="MonoDetour.HookGen" Version="0.7.3" />
     <PackageReference Include="PolySharp" Version="1.15.0">
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
+    <PackageReference Include="Silksong.UnityHelper" Version="1.0.1" />
     <PackageReference Include="UnityEngine.Modules" Version="6000.0.50" IncludeAssets="compile" />
-    <PackageReference Include="Silksong.GameLibs" Version="*-*" PrivateAssets="all" />
     <PackageReference Include="CSharpier.MsBuild" Version="1.1.2" PrivateAssets="all" />
   </ItemGroup>
+  <ItemGroup Condition="!$(UseLocalRefs) Or !Exists($(ManagedFolder))">
+    <PackageReference Include="Silksong.GameLibs" Version="*-*" PrivateAssets="all" />
+  </ItemGroup>
+  <ItemGroup Condition="$(UseLocalRefs) And Exists($(ManagedFolder))">
+    <Reference Include="Assembly-CSharp">
+      <HintPath>$(ManagedFolder)\Assembly-CSharp.dll</HintPath>
```

Comment:
I think you can kill many birds with one stone by just doing `$(ManagedFolder)\*.dll` or similar. Should also set up a publicizer on these so we don't have to reflect stuff in gamelibs Refer https://github.com/Hamunii/BepInExTemplateBase/blob/ee3cc1ae/content/BepInExModTemplate/Directory.Build.targets#L44-L77